grammar/TypeQL.g4

```diff
@@ -32,18 +32,18 @@ eof_schema_rule       :   schema_rule      EOF ;
 // TYPEQL QUERY LANGUAGE ========================================================
 
 query                 :   query_define      |   query_undefine
-                      |   query_insert      |   query_delete_or_update
+                      |   query_update      |   query_insert      |   query_delete
                       |   query_match       |   query_match_aggregate
                       |   query_match_group |   query_match_group_agg           ;
 
 query_define          :   DEFINE      definables  ;
 query_undefine        :   UNDEFINE    definables  ;
 
+query_update          :   MATCH       patterns      DELETE  variable_things
+                                                    INSERT  variable_things     ;
```

Comment:
I considered also `query_delete` followed by `INSERT variable_things`, but this reads better in my opinion.

Reply:
I actually think the `query_delete` followed by `INSERT` is better so that we end up with 1 definition of `delete`.
Then i'd also put the `update` right after the `insert` and before the `delete` definition.
The same order should be reflected in the top `query : ...`